### ISSUE TITLE
Custom DMG Palette & Game Boy Color BIOS - Per-game palettes don't work

### ISSUE THREAD

**Issue by bryc** [Drenn1/GameYob#157]
GameYob is supports GBC bios color palettes as a feature - but it doesn't support the **per-game palettes**. Based on [this article](https://tcrf.net/Game_Boy_Color_Bootstrap_ROM) about the GBC BIOS, there are specific per-game palettes used for different games. 

For example, Kirby games, Super Mario Land and Battletoads in Ragnarok's World all have special palettes defined in the GBC bios.

It would be nice if there was an option to use those palettes.

Also, a second slightly related issue: _Customizable DMG palette instead of just B&W_. 
Basically allow the user to set the RGB values for BG, OBJ0 and OBJ1 in the config.

**Comment by Drenn1** [comment]
First of all, I'm assuming you're talking about gameyob DS.

Per-game palettes are supported through the bios. Simply don't press any buttons while the bios is booting up. That being said, I haven't tested the feature in a while, but I would be surprised if it was broken in v0.5.2.

**Comment by bryc** [comment]
Yes, I'm talking about the DS version, sorry. And I'm using 0.5.2. 

It seems that certain games are just missing palettes, like _Battletoads in Ragnarok's World_. Super Mario Land and Kirby 1 seem to have theirs.

And as for a custom default DMG palette, could that be added?

I'd love to give a classic green/red tint to the default grey palette like this:

![image](https://cloud.githubusercontent.com/assets/1408749/23344665/57cfc6e8-fc4e-11e6-896e-9fa0e42090d8.png)
Instead of
![image](https://cloud.githubusercontent.com/assets/1408749/23344724/428fb364-fc4f-11e6-9fe9-297ca4b061c7.png)

Even if just the raw GBR15 color values in the gameyob.ini, it'd be nice!

**Comment by Drenn1** [comment]
I'd suggest trying the bios with that game in bgb as well. If the coloration doesn't work in bgb either, it's probably not my fault.

Alternate palette options would be a sensible thing to add, but potential new features will likely end up in backlog hell.

**Comment by bryc** [comment]
Ah, it doesn't work in bgb for the (U) version of the game but the (E) version works as intended. I guess it's a bug on bgb's part. 

Though definitely consider the palette option if the opportunity shows itself! Gameyob has a lot going for it, probably one of the best GB experiences outside of a real GB or SGB.

**Comment by Drenn1** [comment]
According to the tcrf page, only the european version of battletoads is included in the list.

**Comment by bryc** [comment]
Yeah.. I didn't pay any attention to that, but its true. The (E) version has Nintendo as the publisher, while (U) has Tradewest/Midway as the publisher. I read into it more and it seems only Nintendo-published games get the palette swap.

So I guess Gambatte is the one with incorrect behavior here. I'm surprised that the one game I end up copying over to test happens to be the one edge case. Closing this issue then.
